src/methods/to.js

```diff
@@ -719,37 +719,27 @@ export class ToImplementation {
 
   ToStringValue(realm: Realm, val: Value): Value {
     if (val.getType() === StringValue) return val;
-    let str;
-    if (typeof val === "string") {
-      str = val;
-    } else if (val instanceof NumberValue) {
-      str = val.value + "";
-    } else if (val instanceof UndefinedValue) {
-      str = "undefined";
-    } else if (val instanceof NullValue) {
-      str = "null";
-    } else if (val instanceof SymbolValue) {
-      throw realm.createErrorThrowCompletion(realm.intrinsics.TypeError);
-    } else if (val instanceof BooleanValue) {
-      str = val.value ? "true" : "false";
-    } else if (val instanceof ObjectValue) {
+    if (val instanceof ObjectValue) {
       let primValue = this.ToPrimitiveOrAbstract(realm, val, "string");
       if (primValue.getType() === StringValue) return primValue;
-      str = this.ToStringPartial(realm, primValue);
+      let str = this.ToStringPartial(realm, primValue);
```

Comment:
This looks like a candidate for ToStringValue.

Reply:
👍